anxcloud/schema_virtual_server.go

```diff
@@ -18,17 +18,29 @@ func schemaVirtualServer() map[string]*schema.Schema {
 			ForceNew:    true,
 			Description: "Location identifier.",
 		},
+		"template": {
```

Comment:
not sure if I like the `$template_name:$template_version` format or if two attributes would be better. Is it common like this in terraform?

Reply:
AFAIK field types can't be optional.

Reply:
can't find anything for that either

> I think the second version would be too cluttered with template_id and template_type also present as an alternative.

I'd expect the build to only be set in a few cases, but the template a lot - that's why I'm hesitant to go for the object form. You have a point with that, though.. `template` and `template_build`? Still cluttered, but looks very ok in most cases and is "only" cluttered in docs?

Reply:
@HaveFun83 tie breaker please, virtual machines with named templates:

```hcl
template = {
    name = "Debian 11"
    build = "42" // defaults to "latest"
}
```

--vs--

```hcl
template = "Debian 11"
template_build = "42" // defaults to "latest"
```

--vs--

```hcl
template_name = "Debian 11"
template_build = "42" // defaults to "latest"
```

--vs--

```hcl
template = "Debian 11:42" // defaults to latest when only "Debian 11" is written
```